telemetry/telemetrybuffer_test.go

```diff
@@ -0,0 +1,157 @@
+package telemetry
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const telemetryConfig = "azure-vnet-telemetry.config"
+
+func createTBServer(t *testing.T) (*TelemetryBuffer, func()) {
+	tbServer := NewTelemetryBuffer()
+	err := tbServer.StartServer()
+	require.NoError(t, err)
+
+	return tbServer, func() {
+		tbServer.Close()
+		err := tbServer.Cleanup(FdName)
+		require.Error(t, err)
+	}
+}
+
+func TestStartServer(t *testing.T) {
+	_, closeTBServer := createTBServer(t)
+	defer closeTBServer()
+
+	secondTBServer := NewTelemetryBuffer()
+	err := secondTBServer.StartServer()
+	require.Error(t, err)
+}
+
+func TestConnect(t *testing.T) {
+	_, closeTBServer := createTBServer(t)
+	defer closeTBServer()
+
+	tbClient := NewTelemetryBuffer()
+	err := tbClient.Connect()
+	require.NoError(t, err)
+
+	tbClient.Close()
+}
+
+func TestServerConnClose(t *testing.T) {
+	tbServer, closeTBServer := createTBServer(t)
+	defer closeTBServer()
+
+	tbClient := NewTelemetryBuffer()
+	err := tbClient.Connect()
+	require.NoError(t, err)
+	defer tbClient.Close()
+
+	tbServer.Close()
+
+	b := []byte("testdata")
+	_, err = tbClient.Write(b)
+	require.Error(t, err)
+}
+
+func TestClientConnClose(t *testing.T) {
+	_, closeTBServer := createTBServer(t)
+	defer closeTBServer()
+
+	tbClient := NewTelemetryBuffer()
+	err := tbClient.Connect()
+	require.NoError(t, err)
+	tbClient.Close()
+}
+
+func TestWrite(t *testing.T) {
+	_, closeTBServer := createTBServer(t)
+	defer closeTBServer()
+
+	tbClient := NewTelemetryBuffer()
+	err := tbClient.Connect()
+	require.NoError(t, err)
+	defer tbClient.Close()
+
+	tests := []struct {
+		name    string
+		data    []byte
+		want    int
+		wantErr bool
+	}{
+		{
+			name:    "write",
+			data:    []byte("testdata"),
+			want:    len("testdata") + 1, // +1 due to Delimiter('\n)
+			wantErr: false,
+		},
+		{
+			name:    "write zero data",
+			data:    []byte(""),
+			want:    1, // +1 due to Delimiter('\n)
+			wantErr: false,
+		},
+	}
+
+	for _, test := range tests {
+		tt := test
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := tbClient.Write(tt.data)
+			if tt.wantErr {
+				assert.Error(t, err)
+				return
+			}
+			assert.NoError(t, err)
```

Comment:
```suggestion
			if tt.wantErr {
				require.Error(t, err)
			}
			require.NoError(t, err)
```

Reply:
While still it still needs `return` in `if` statement. It not, it runs `require.NoError(t, err)`